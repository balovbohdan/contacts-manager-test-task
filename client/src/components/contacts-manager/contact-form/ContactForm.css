.main {}

.inputWrapper {
    margin: 20px 0 40px;
}

.foot {
    display: flex;
    align-items: center;
    justify-content: flex-end;
}