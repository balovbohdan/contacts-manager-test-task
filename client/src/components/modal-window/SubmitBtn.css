.submitBtn {
    cursor: pointer;
    padding: 10px 15px;
    transition: var(--trans);
    border-radius: var(--border-radius);
    background-color: rgba(0, 0, 0, .2);
}

.submitBtn:hover {
    background-color: rgba(0, 0, 0, .3);
}

.submitBtn__disabled {
    cursor: not-allowed;
}

.submitBtn__disabled:hover {
    background-color: rgba(0, 0, 0, .2);
}