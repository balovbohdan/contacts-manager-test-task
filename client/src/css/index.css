@import url(reset.css);
@import url(fonts.css);
@import url(titles.css);
@import url(constants.css);
@import url(main.css);