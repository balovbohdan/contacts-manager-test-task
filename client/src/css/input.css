.label {
    font-size: .9rem;
}

.input {
    width: 100%;
    padding: 5px 0;
    border-bottom: 2px solid var(--color-yellow);

    font-size: 1.5rem;
    color: var(--color-white);
}

@media (max-width: 600px) {
    .input {
        font-size: 1.2rem;
    }

    .label {
        font-size: .8rem;
    }
}

@media (max-width: 400px) {
    .input {
        font-size: 1.1rem;
    }

    .label {
        font-size: .7rem;
    }
}