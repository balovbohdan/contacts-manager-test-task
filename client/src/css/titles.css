h1, h2, h3, h4, h5, h6 {
    font-size: 100%;
    font-family: var(--font-main);
}

h1 {
    font-size: 40px;
}

h2 {
    font-size: 35px;
}

h3 {
    font-size: 30px;
}

h4 {
    font-size: 25px;
}

h5 {
    font-size: 20px;
}

h6 {
    font-size: 17px;
}

@media (max-width: 550px) {
    h1 {
        font-size: 35px;
    }

    h3 {
        font-size: 25px;
    }
}

@media (max-width: 400px) {
    h1 {
        font-size: 30px;
    }

    h3 {
        font-size: 20px;
    }
}