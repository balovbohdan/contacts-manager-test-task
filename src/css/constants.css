:root {
    --color-red: #FB4B4E;
    --color-blue: #16324F;
    --color-white: #F3F7F0;
    --color-yellow: #FDE74C;
    --color-light-blue: #236098;

    --font-main: Roboto, serif;
    --font-marker: 'Permanent Marker', serif;

    --border-radius: 3px;
    --header-height: 60px;
    --trans: all .12s ease-in;
}