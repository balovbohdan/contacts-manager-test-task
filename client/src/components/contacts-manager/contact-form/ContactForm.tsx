import * as React from 'react';

import img from '@img/user.svg';
import inputCss from '@css/input.css';
import {SubmitBtn} from '@components/modal-window';

import * as T from './types';
import css from './ContactForm.css';
import * as utils from './utils';

const {useState, useEffect, useCallback} = React;

export const ContactForm = (props:T.Props) => {
    const def = props.def || {};

    const [isValid, setIsValid] = useState(true);

    const [tip, setTip] = useState(def.tip || '');
    const [name, setName] = useState(def.name || '');
    const [phone, setPhone] = useState(def.phone || '');

    const onKeyDown = ({key}) => key === 'Enter' && submit();
    const onTipChange = ({target}) => setTip(target.value || '');
    const onNameChange = ({target}) => setName(target.value || '');
    const onPhoneChange = ({target}) => setPhone(target.value || '');

    const getData = ():T.FormData => ({ tip, img, name, phone });
    const submit = utils.createSubmit(isValid, props.submit, getData);

    const updateIsValid = useCallback(() => {
        const formData = getData();
        const isValid = utils.checkFormData(formData);

        setIsValid(isValid);
    }, [name, phone]);

    useEffect(() => {
        updateIsValid();
    }, [name, phone]);

    return (
        <div className={css.main}>
            <div className={css.inputWrapper}>
                <label className={inputCss.label}>Name:</label>
                <input
                    type='text'
                    maxLength={30}
                    autoFocus={true}
                    defaultValue={name}
                    onKeyDown={onKeyDown}
                    onChange={onNameChange}
                    className={inputCss.input}/>
            </div>
            <div className={css.inputWrapper}>
                <label className={inputCss.label}>Phone:</label>
                <input
                    type='text'
                    maxLength={30}
                    defaultValue={phone}
                    onKeyDown={onKeyDown}
                    onChange={onPhoneChange}
                    className={inputCss.input}/>
            </div>
            <div className={css.inputWrapper}>
                <label className={inputCss.label}>Tip:</label>
                <input
                    type='text'
                    maxLength={30}
                    defaultValue={tip}
                    onKeyDown={onKeyDown}
                    onChange={onTipChange}
                    className={inputCss.input}/>
            </div>
            <Foot submit={submit} isDataValid={isValid}/>
        </div>
    );
};

const Foot = ({submit, isDataValid}:T.FootProps) =>
    <div className={css.foot}>
        <SubmitBtn submit={submit} isDataValid={isDataValid}/>
    </div>;