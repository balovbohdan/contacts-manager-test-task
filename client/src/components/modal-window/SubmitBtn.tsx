import * as React from 'react';
import classnames from 'classnames';

import * as css from './SubmitBtn.css';

type Props = {
    submit:()=>void;
    isDataValid:boolean;
};

export const SubmitBtn = ({submit, isDataValid}:Props) => {
    const doSubmit = () => isDataValid && submit();

    const className = classnames({
        [css.submitBtn]: true,
        [css.submitBtn__disabled]: !isDataValid
    });

    return <div onClick={doSubmit} className={className}>Submit</div>;
};