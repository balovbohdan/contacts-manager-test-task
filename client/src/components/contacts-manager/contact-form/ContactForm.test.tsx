import * as React from 'react';
import {mount, shallow} from 'enzyme';

import {ContactForm} from './ContactForm';
import * as css from '../../modal-window/SubmitBtn.css';

describe('ContactForm', () => {
    it('Should match snapshot', () => {
        const props = {
            submit: jest.fn()
        };

        const form = shallow(<ContactForm {...props}/>);

        expect(form).toMatchSnapshot();
    });

    it('Should react on submit btn click', () => {
        const props = {
            submit: jest.fn(),
            def: {
                tip: 'tip',
                name: 'name',
                phone: '+380961514231'
            }
        };

        const form = mount(<ContactForm {...props}/>);

        form.find(`.${css.submitBtn}`).simulate('click');
        expect(props.submit).toBeCalledTimes(1);
    });
});