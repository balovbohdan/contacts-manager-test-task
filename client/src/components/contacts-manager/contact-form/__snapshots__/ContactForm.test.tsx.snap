// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Should match snapshot 1`] = `
<div
  className="main"
>
  <div
    className="inputWrapper"
  >
    <label>
      Name:
    </label>
    <input
      autoFocus={true}
      defaultValue=""
      maxLength={30}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
    />
  </div>
  <div
    className="inputWrapper"
  >
    <label>
      Phone:
    </label>
    <input
      defaultValue=""
      maxLength={30}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
    />
  </div>
  <div
    className="inputWrapper"
  >
    <label>
      Tip:
    </label>
    <input
      defaultValue=""
      maxLength={30}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
    />
  </div>
  <Foot
    isDataValid={true}
    submit={[Function]}
  />
</div>
`;
